auth/auth.go

```diff
@@ -31,6 +31,7 @@ type UserIdentity struct {
 	CurrentUser  bool
 	AuthUsername string // Username matched in privileges system
 	AuthHostname string // Match in privs system (i.e. could be a wildcard)
+	IsRole       bool
```

Comment:
How about defining a `RoleIdentity` ?
```
type RoleIdentity struct {
    Username string
    Hostname string
}
```

Reply:
I plan to reuse `UserIdentity` and `CreateUserStmt` for `CreateRoleStmt`.